E2E Tests: Don't pre-connect for suites where it is not needed (#17470)
Automattic/jetpack: tests/e2e/README.md

- [jest-puppeteer](https://github.com/smooth-code/jest-puppeteer) – provides all required configuration to run tests using Puppeteer
- [expect-puppeteer](https://github.com/smooth-code/jest-puppeteer/tree/master/packages/expect-puppeteer) – assertion library for Puppeteer

> Tests are kept in `tests/e2e/specs` folder. Every file represents a test suite, which is designed around specific feature under test. Most of the tests relies on active Jetpack Connection, so we do connect a site before running actual test suite. It's logic could be found in [`setup-env#maybePreConnect`](./lib/setup-env.js) function. For test suites where pre-connection is not need, it could be disabled by setting `SKIP_CONNECT` env var to false. Check [`connection.test.js`](./specs/connection.test.js) for example use.

Reviewer: ```suggestion
Tests are kept in `tests/e2e/specs` folder. Every file represents a test suite, which is designed around specific feature under test. Most of the tests rely on an active Jetpack Connection, so we connect a site before running the actual test suite. Its logic can be found in the [`setup-env#maybePreConnect`](./lib/setup-env.js) function. For test suites where pre-connection is not needed, it can be disabled by setting `SKIP_CONNECT` env var to false. Check [`connection.test.js`](./specs/connection.test.js) for example use.
```
